common/src/main/java/com/wynntils/functions/MinecraftFunctions.java

```diff
@@ -12,28 +12,28 @@ public class MinecraftFunctions {
     public static class XFunction extends Function<Integer> {
         @Override
         public Integer getValue(String argument) {
-            return McUtils.player().getBlockX();
+            return McUtils.player() == null ? null : McUtils.player().getBlockX();
```

Comment:
How can McUtils.player() be null? I thought it was set as soon as you logged on to the server.

Reply:
For some reason, this was called on feature init. As I am writing this, because this is what the render preview used to be. And it used to be a static field..., so this was needed to be calculated. I fixed that so this is not needed anymore technically. I am not sure if there is value to keeping this?

Reply:
(Also, functions do not fail gracefully :))

Reply:
I thought I actually wrote a rationale about skipping functions: They are so simple I did not expect them to fail, and if they do, the failure will propagate to the `/function` command and/or InfoBoxOverlay, which will fail gracefully.

But it might be a point in disabling individual functions. If one function is behaving bad, we do not have to bring down the entire InfoBoxOverlay. I'll fix that.

Reply:
This case was more of bad use of functions than them being bad, but I think there is still value to disabling functions.

Reply:
And no, if it's not needed we should not keep it. I've been trying to kill code that does null checks for player() elsewhere. If we have such code, people start thinking it's needed and add it in some places but not other. We must be able to trust this value, period.

Also, functions should perhaps be blocked by the FunctionManager if we are not connected to a wynncraft server, just like events are not posted. That way we can definitely not do anything wrong with this.

Reply:
Reverted this